File: tests/statistical_tests/spot_checks/proportion-one-sample.test.ts
import { stats } from "../../../src/dataframe/mod.ts";

Deno.test("One-Sample Proportion Test Spot Check", () => {
  console.log("\n📊 One-Sample Proportion Test");
  console.log("-".repeat(80));

  const data = [true, true, false, true, true, true, false, true, true, false];
  const p0 = 0.5;

  const result = stats.test.proportion.oneSample({
    data,
    popProportion: p0,
    alternative: "two-sided",
  });

  const successes = data.filter((x) => x).length;
  const n = data.length;

  console.log("Data:", data);
  console.log("Successes:", successes, "/", n);
  console.log("p0 (null hypothesis):", p0);
  console.log("Test statistic:", result.test_statistic.value);
  console.log("p-value:", result.p_value);
  console.log("CI:", [
    result.confidence_interval.lower,
    result.confidence_interval.upper,
  ]);
});

console.log("\n" + "=".repeat(80));
console.log("ONE-SAMPLE PROPORTION TEST SPOT CHECK");
console.log("=".repeat(80));

File: tests/statistical_tests/spot_checks/proportion-one-sample.test.R
#!/usr/bin/env Rscript

cat("\n📊 One-Sample Proportion Test\n")
cat(paste(rep("-", 80), collapse = ""), "\n")

successes <- 7
n <- 10
p0 <- 0.5

# One-sample proportion test
result <- prop.test(successes, n, p = p0, alternative = "two.sided", conf.level = 0.95, correct = TRUE)

cat("Successes:", successes, "/", n, "\n")
cat("p0 (null hypothesis):", p0, "\n")
cat("Chi-squared statistic:", result$statistic, "\n")
cat("p-value:", result$p.value, "\n")
cat("CI lower:", result$conf.int[1], "\n")
cat("CI upper:", result$conf.int[2], "\n")
cat("Sample estimate:", result$estimate, "\n")

cat("\n", paste(rep("=", 80), collapse = ""), "\n")
cat("ONE-SAMPLE PROPORTION TEST SPOT CHECK\n")
cat(paste(rep("=", 80), collapse = ""), "\n")
